rust_src/src/data.rs

```diff
@@ -49,3 +52,17 @@ pub fn indirect_function_lisp(object: LispObject, _noerror: LispObject) -> LispO
     }
     return result;
 }
+
+/// Return SYMBOL's value.  Error if that is void.  Note that if
+/// `lexical-binding' is in effect, this returns the global value
+/// outside of any lexical scope.
+#[lisp_fn]
+pub fn symbol_value(symbol: LispObject) -> LispObject {
+    unsafe {
```

Comment:
I'd like to get rid of this unsafe block/port `find_symbol_value` in the near future. However, there are a lot of fiddly bits having to do with `Lisp_Symbol` and the bitfield at the top. Right now there is a way to tell whether or not a given symbol is an alias but no general way to pull the enum value out of the bitfield.

Also, abstracting that redirect enum would be nice as it would eliminate the need for unsafe blocks when matching on it. Is there a precedent in the project for something like that?